### PR TITLE
Fix WDM-KS buffer position alignment

### LINKED ISSUE #763: WDM-KS audio is choppy with some values of suggestedLatency
**Describe the bug**
When using the WDM-KS host API, audio is choppy with some values of `suggestedLatency`. The distribution of good and bad values seems random. This is not specific to small values.

**To Reproduce**
1. Change `examples/paex_sine.c`:
   - Set `device` to the index of a WDM-KS device.
      - The hardware I'm testing with is an ASUS Xonar U7 using the Microsoft standard `usbaudio2.sys` driver (WaveRT).
   - Try various reasonable values for `suggestedLatency`.
3. Run the example.

**Expected behavior**
Smooth audio. In my testing this happens for some values of suggestedLatency, for example `0.030` or `0.031`, but not `0.029` nor `0.100`.

This is an example debug output with `0.031` which works:

```
 0.067: OpenStream:sampleRate = 44100.000000
 0.067: OpenStream:framesPerBuffer = 64
 0.067: Pin create result = 0x00000491
 0.067: Pin create result = 0x00000491
 0.067: Pin create result = 0x00000491
 0.067: Pin create result = 0x00000491
 0.067: Pin create result = 0x00000491
 0.067: Pin create result = 0x00000491
 0.068: Pin create result = 0x00000491
 0.068: Pin create result = 0x00000491
 0.070: Pin create result = 0x00000000
 0.070: Render pin frames: 0
 0.070: Output frames chosen:1367
 0.070: PinGetBuffer: SubType_kNotification
 0.070: Output buffer start = 000001CF37D60000, size = 16404, membarrier = 0
 0.070: BytesPerInputFrame = 0
 0.071: BytesPerOutputFrame = 6
 0.071: Failed to register position register
 0.071: Failed to register rendering position register, using PinGetAudioPositionViaIOCTLRead
Latency: 0.030998s
 0.072: In  buffer len: 0.000 ms
 0.072: Out buffer len: 61.995 ms
 0.072: Timeout = 496 ms
 0.074: PreparePinsForStart = 0
 0.074: Processing thread started!
Play for 5 seconds.
```

**Actual behavior**
Choppy audio with many discontinuities per second. This only happens with some values of `suggestedLatency` but not others, with no apparent logic.

This is an example debug output with `0.029` which is choppy:

```
 0.072: OpenStream:sampleRate = 44100.000000
 0.073: OpenStream:framesPerBuffer = 64
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.073: Pin create result = 0x00000491
 0.075: Pin create result = 0x00000000
 0.076: Render pin frames: 0
 0.076: Output frames chosen:1278
 0.076: PinGetBuffer: SubType_kNotification
 0.076: Output buffer start = 0000020AD3CB0000, size = 15336, membarrier = 0
 0.076: BytesPerInputFrame = 0
 0.076: BytesPerOutputFrame = 6
 0.076: Failed to register position register
 0.076: Failed to register rendering position register, using PinGetAudioPositionViaIOCTLRead
Latency: 0.028980s
 0.077: In  buffer len: 0.000 ms
 0.077: Out buffer len: 57.959 ms
 0.077: Timeout = 464 ms
 0.079: PreparePinsForStart = 0
 0.079: Processing thread started!
Play for 5 seconds.
```

**Desktop:**
 - OS Version: Windows 11 21H2 22000.1455
 - PortAudio version: latest master (bbe2b5ad55229c13c4d7cd54b5dcd78511c70a4b)
 - Host API: WDM-KS

**Additional context**

I noticed #752 from @feoff3 which seemed promising, but I can still reproduce with that PR. This is unsurprising because that PR only touches the capture code path, but the example code does playback.

### PR DESCRIPTION
**Pull request by dechamps**
The previous code is only correct if `bytesPerFrame` is a power of two. If it's not (e.g. 2-channel 24-bit = 6 bytes per frame), then the computed buffer position is corrupted. This can lead the code to use the wrong buffer half, resulting in glitchy audio.

Fixes #763

### CODE REVIEW

**Review by RossBencina — APPROVED** [review]
Looks good to me. I approve.

Just wondering whether a comment like `/* note bytesPerFrame can be non-power-of-two */` might be an improvement.

**Comment by RossBencina** [comment]
Thanks for tracking this down. Great to have our WDM/KS support inch closer to solidity. I'll wait for Phil's review before merging.

**Review by philburk — APPROVED** [review]
Thanks!

**Pull request merged**
